packages/flutter/test/material/dialog_test.dart

```diff
@@ -607,4 +607,43 @@ void main() {
     expect(find.text('0'), findsNothing);
     expect(find.text('1'), findsNothing);
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/28505.
+  testWidgets('showDialog only gets Theme from context on the first call', (WidgetTester tester) async {
+    Widget buildFrame(Key builderKey) {
+      return MaterialApp(
+        home: Center(
+          child: Builder(
+            key: builderKey,
+            builder: (BuildContext outerContext) {
+              return RaisedButton(
+                onPressed: () {
+                  showDialog<void>(
+                    context: outerContext,
+                    builder: (BuildContext innerContext) {
+                      return const AlertDialog(title: Text('Title'));
+                    },
+                  );
+                },
+                child: const Text('Show Dialog'),
+              );
+            },
+          ),
+        ),
+      );
+    }
+
+    // First time build.
+    await tester.pumpWidget(buildFrame(UniqueKey()));
+
+    // Open the dialog.
+    await tester.tap(find.byType(RaisedButton));
+    await tester.pumpAndSettle();
+
+    // Second time build (deactivate old context).
```

Comment:
It would be useful to say: 
Force the Builder to be recreated (new key) which causes outerContext to be deactivated. If showDialog()'s  implementation were to refer to outerContext again, it would crash.

Reply:
Done.